chevah/compat/tests/elevated/test_system_users.py

```diff
@@ -244,6 +244,84 @@ def test_authenticateWithUsernameAndPassword_good(self):
         else:
             self.assertIsNotNone(token)
 
+    @conditionals.onOSFamily('posix')
+    def test_authenticateWithUsernameAndPassword_passwd_valid(self):
+        """
+        On most OS system password is not stored in the passwd file so even
+        if we pass the correct pass it will return None to inform that
+        password is not here.
+        """
+        result = system_users._checkPasswdFile(
+            username=TEST_ACCOUNT_USERNAME,
+            password=TEST_ACCOUNT_PASSWORD,
+            )
+
+        if self.os_name in ['aix', 'hpux']:
+            # On AIX and HPUX password is here.
+            self.assertTrue(result)
+        else:
+            # Not here.
+            self.assertIsNone(result)
+
+    @conditionals.onOSFamily('posix')
+    def test_authenticateWithUsernameAndPassword_passwd_invalid(self):
+        """
+        When an invalid password is provided, on most OS system password is not
+        stored in the passwd file it return None to inform that
+        password is not here.
+        On systems with passwd it return False.
+        """
+        result = system_users._checkPasswdFile(
+            username=TEST_ACCOUNT_USERNAME, password='')
+
+        if self.os_name in ['aix', 'hpux']:
+            # On AIX and HPUX invalid passwords are not accepted.
+            self.assertFalse(result)
+        else:
+            # On all other OSs password is not here.
+            self.assertIsNone(result)
+
+    @conditionals.onOSFamily('posix')
+    def test_authenticateWithUsernameAndPassword_shadow(self):
+        """
+        Check successful call to authenticateWithUsernameAndPassword.
+        """
+        if self.os_name in ['aix', 'hpux']:
+            # AIX and HPUX has no shadow
+            raise self.skipTest()
+
+        result = system_users._checkShadowFile(
+            username=TEST_ACCOUNT_USERNAME,
+            password=TEST_ACCOUNT_PASSWORD,
+            )
+
+        if self.os_name in ['aix', 'hpux', 'osx']:
```

Comment:
thanks... test should not be skipped, but rather check that shadow is not supported